### ISSUE TITLE
v2 vars_prompt does not have access to inventory facts

### ISSUE THREAD

**Issue by drfickle** [ansible/ansible#15290]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION

```
ansible 2.0.1.0
  config file = /home/sfox/.ansible.cfg
  configured module search path = /home/sfox/src/sysops/ops/branches/ansible-2.0/ansible/library
```
##### CONFIGURATION

```
library = /home/sfox/src/sysops/ops/branches/ansible-2.0/ansible/library
forks          = 20
sudo_flags = -H -i
jinja2_extensions = jinja2.ext.do,jinja2.ext.i18n,jinja2.ext.loopcontrols,jinja2.ext.with_
ansible_managed = Ansible managed: {file} modified on %Y-%m-%d %H:%M:%S by {uid} on {host}
action_plugins     = /usr/share/ansible_plugins/action_plugins:/home/sfox/src/sysops/ops/branches/ansible-2.0/ansible/plugins/action
callback_plugins   = /usr/share/ansible_plugins/callback_plugins:/home/sfox/src/sysops/ops/branches/ansible-2.0/ansible/plugins/callback
connection_plugins = /usr/share/ansible_plugins/connection_plugins:/home/sfox/src/sysops/ops/branches/ansible-2.0/ansible/plugins/connection
lookup_plugins     = /usr/share/ansible_plugins/lookup_plugins:/home/sfox/src/sysops/ops/branches/ansible-2.0/ansible/plugins/lookup
vars_plugins       = /usr/share/ansible_plugins/vars_plugins:/home/sfox/src/sysops/ops/branches/ansible-2.0/ansible/plugins/vars
filter_plugins     = /usr/share/ansible_plugins/filter_plugins:/home/sfox/src/sysops/ops/branches/ansible-2.0/ansible/plugins/filter
nocows = 1
```
##### OS / ENVIRONMENT

From: Fedora 23
Managing: Ubuntu Trusty (14.04)
##### SUMMARY

Prompted variables which have defaults originating from inventory-supplied facts no longer work in 2.0
##### STEPS TO REPRODUCE

Using this inventory file

```
test_host ansible_host=10.0.0.208 version=11.0
```

and this playbook

```
- name: My Playbook
  hosts: test_host
  gather_facts: no

  vars_prompt:                                                                                                                                                                                                     
    - name: from_ver
      prompt: "From which version are you upgrading? (from_ver)"
      default: "{{ version }}"
      private: no

  tasks:
    - debug: var=version
```

the task displays the value of the version variable without the vars_prompt section enabled, but not with it in.
##### EXPECTED RESULTS

```
PLAY [My Playbook] *************************************************************

TASK [debug] *******************************************************************
ok: [test_host] => {
    "version": 11.0
}
```
##### ACTUAL RESULTS

```
ERROR! 'version' is undefined
```

This also fails with the devel branch.

**Comment by nitzmahone** [comment]
Thanks for the report, but I suspect this is a "won't fix" case- this shouldn't have worked at all in 1.9. vars_prompt is a play-level directive (note that you're not prompted for each host)- the values collected by the prompt are only good for the duration of the play, and will be the same for each host (as they're not copied to the hosts vars collection, they're stored with the play). 

Even for 1.9, which host's vars were we looking at for a play-level case if the play targets more than one? This was probably one of the cases that got fixed in 2.0 with the new VariableManager more strictly enforcing variable scoping.

Note that should be possible to make this work for the sample you provided by using:

```
default: "{{ hostvars['test_host']['version'] }}"
```

because you're specifically saying which host's version var you want to use, but given that this was a case of "worked accidentally and unpredictably" in 1.x, I wouldn't expect that 2.x will change to replicate that behavior.

**Comment by drfickle** [comment]
That makes sense. For my case it was used across servers for which that variable was always the same. Thank you for the explanation and your time looking into it.
